src/common/entity/compute_state_display.ts

```diff
@@ -126,7 +126,7 @@ export const computeStateDisplayFromEntityAttributes = (
         );
         return formatDateTime(date, locale);
       }
-      if (attributes.has_date) {
+      if (domain === "date" || attributes.has_date) {
         date = new Date(attributes.year, attributes.month - 1, attributes.day);
```

Comment:
The `date` entity will not have these attributes, so, in that case, we need to create a date from the state string.

I don't think this part is even needed for this integration, date parsing done above, seems to be enough?

Reply:
correct, removing